common/src/KokkosFFT_utils.hpp

```diff
@@ -243,6 +245,44 @@ constexpr Kokkos::Array<T, N> to_array(std::array<T, N>&& a) {
   return to_array_rvalue(std::move(a), std::make_index_sequence<N>());
 }
 
+template <typename T>
+T safe_multiply(T a, T b) {
+  if constexpr (std::is_integral_v<T>) {
+    if constexpr (std::is_signed_v<T>) {
+      if (a != 0) {
+        if (a > 0) {
+          if ((b > 0 && a > std::numeric_limits<T>::max() / b) ||
+              (b < 0 && b < std::numeric_limits<T>::min() / a)) {
+            throw std::overflow_error("Integer multiplication overflow");
+          }
+        } else {  // a < 0
+          if ((b > 0 && a < std::numeric_limits<T>::min() / b) ||
+              (b < 0 && a != 0 && -a > std::numeric_limits<T>::max() / -b)) {
```

Comment:
```suggestion
              (b < 0 && -a > std::numeric_limits<T>::max() / -b)) {
```
This comparison is useless since we are already in the `if (a != 0)` branch